libs/utils/website/src/lib/pages/subscribe-page.tsx

```diff
@@ -0,0 +1,126 @@
+import {ApiV1, AuthTokenStorageKey, SubscribeContainer} from '@wepublish/website'
+import {setCookie} from 'cookies-next'
+import {NextPageContext} from 'next'
+import getConfig from 'next/config'
+import {useRouter} from 'next/router'
+import {ComponentProps} from 'react'
+import {ssrAuthLink} from '../auth-link'
+import {getSessionTokenProps} from '../get-session-token-props'
+
+type SubscribePageProps = Omit<
+  ComponentProps<typeof SubscribeContainer>,
+  'failureURL' | 'successURL'
+>
+
+export function SubscribePage(props: SubscribePageProps) {
+  const {
+    query: {
+      memberPlanBySlug,
+      additionalMemberPlans,
+      firstName,
+      mail,
+      lastName,
+      deactivateSubscriptionId
+    }
+  } = useRouter()
+
+  return (
+    <SubscribeContainer
+      {...props}
+      defaults={{
+        email: mail as string | undefined,
+        firstName: firstName as string | undefined,
+        name: lastName as string | undefined,
+        memberPlanSlug: memberPlanBySlug as string | undefined,
+        ...props.defaults
+      }}
+      filter={memberPlans => {
+        const parentFiltered = props.filter?.(memberPlans) ?? memberPlans
+
+        const preselectedMemberPlan = parentFiltered.find(({slug}) => slug === memberPlanBySlug)
+
+        if (additionalMemberPlans === 'upsell' && preselectedMemberPlan) {
+          return parentFiltered.filter(
+            memberPlan =>
+              memberPlan.amountPerMonthMin >= preselectedMemberPlan.amountPerMonthMin ||
+              memberPlan === preselectedMemberPlan
+          )
+        }
+
+        return preselectedMemberPlan && additionalMemberPlans !== 'all'
+          ? [preselectedMemberPlan]
+          : parentFiltered
+      }}
+      deactivateSubscriptionId={
+        props.deactivateSubscriptionId ?? (deactivateSubscriptionId as string | undefined)
+      }
+      failureURL=""
```

Comment:
This connection got lost, but is important. Could we solve it by merging first the SHARE-42 story?

Reply:
SHARE-43? That's the point why this is empty, as SHARE-43 will be merged first.